packages/block-editor/src/components/link-control/index.js

```diff
@@ -301,7 +301,7 @@ function LinkControl( {
 							isURL={ manualLinkEntryTypes.includes(
 								suggestion.type.toLowerCase()
 							) }
-							searchTerm={ inputValue }
+							searchTerm={ inputValue.trim() }
```

Comment:
`trim()` here cannot be removed. 

According to `TextHighlight` component used in `LinkControlSearchItem`, it tries to find the exact string to highlight. 

https://github.com/WordPress/gutenberg/blob/5a5564b4d5cd6fe271ccb02e3f85f4141d7f21b5/packages/components/src/text-highlight/index.js#L11-L24

In other words, if `trim()` isn't here, `TextHighlight` tries to find `"    Hello    "` from "Hello World!" and it doesn't match. 

I thought of `TextHighlight` trimming whitespace, but it can affect the behavior of many other components. So, I think putting trim() here is enough for now.

Reply:
I wonder whether

* We could ensure `inputValue` has already been trimmed in `URLInput` before it's propagated upward?

or 

* We should move the `.trim()` into the `TextHighlight` component?

I note you say:

> I thought of TextHighlight trimming whitespace, but it can affect the behavior of many other components. So, I think putting trim() here is enough for now.

But have you checked whether `TextHighlight` is in use anywhere else because I introduced it fairly recently so I doubt it's widespread. Moreover, I would have thought that the desired functionality would never be to highlight a text string that included whitespace around it. Just thinking out loud, the purpose of the component is to visuall highlight a discrete string. 

Interested in your opinion.

Reply:
I don't think trimming inputValue inside URLInput is a good idea. Then, when users type in trailing/leading spaces, it won't show on the input field. And we might need to manage two versions of `inputValue` in URLInput: one for display, one for networking. 

The second option looks better to me. I checked where `TextHighlight` is used and it was used only once in `LinkControl`. It doesn't affect other behaviors, maybe a storyshot. I need to test it out. 

I didn't write it above, but I also thought of adding `trim` prop to `TextHighlight`. You gave me the answer:

> Just thinking out loud, the purpose of the component is to visuall highlight a discrete string.

Generating `<mark>    Hello    </mark> World` isn't what we want. We want trimmed string around the `<mark>` tag. 

If you don't mind, I'll add `.trim()` to `TextHighlight`.

Reply:
> when users type in trailing/leading spaces, it won't show on the input field.

Good point. Not looking at this in context I didn't consider that. 👍 

> If you don't mind, I'll add .trim() to TextHighlight.

I think this is the way forward. 👍